crates/common/types/transaction.rs

```diff
@@ -406,31 +405,30 @@ impl RLPDecode for Transaction {
     /// B) Non legacy transactions: rlp(Bytes) where Bytes represents the canonical encoding for the transaction as a bytes object.
     /// Checkout [Transaction::decode_canonical] for more information
     fn decode_unfinished(rlp: &[u8]) -> Result<(Self, &[u8]), RLPDecodeError> {
-        if is_encoded_as_bytes(rlp)? {
-            // Adjust the encoding to get the payload
-            let payload = get_rlp_bytes_item_payload(rlp)?;
```

Comment:
I think this pattern is also used for receipt decoding. Maybe we have the same bug there too.

Reply:
Addressed [2548ae7](https://github.com/lambdaclass/ethrex/pull/5139/commits/2548ae7bc08e65f373e8311667c86f56a791c0a6)